### DISCUSSION TITLE
Headless Collaboration

### DISCUSSION THREAD

**Discussion by DarkArc** [zed-industries/zed#34468] [Feature Requests]
### Summary

It would be great if Zed had an option where you could on the command line setup a collaborative editing environment.

Something like:
```
zed --run-collaboration-server <zed-host-id-key> \
    --invite <zed-id-1> --invite <zed-id-2> \
    --invite <zed-user-group-1> --invite <zed-user-group-2> \
    <project-dir>
```

Then those users/people in those groups would be able to connect into the project an edit it remotely without someone having to keep a Zed instance alive at all times.

**Comment by DarkArc** [comment]
This is similar to remote development. Maybe this is possible via a remote connection + share through the GUI (that would certainly be nice, though command line "driver" of some sort also seems nice). I don't have any new enough servers with a new enough glibc to test this at the moment.

**Comment by ConradIrwin** [comment]
We've talked about this! Likely we'd build it into Zed's channels, so you can share a project into a channel and have it remain there even after you leave the call.

**Comment by notpeter** [comment]
This is not something which is currently on our roadmap, but none the less an interesting idea.
